library/test/src/types.rs

```diff
@@ -140,6 +142,28 @@ impl TestDesc {
             }
         }
     }
+
+    pub fn test_mode_string(&self) -> String {
+        if self.ignore {
+            return "ignore".to_string();
+        }
+        match self.should_panic {
+            options::ShouldPanic::Yes | options::ShouldPanic::YesWithMessage(_) => {
+                return "should panic".to_string();
+            }
+            _ => {}
```

Comment:
Might be good to match on `No` explicitly, so we don't forget to update this if we ever add another `Yes` option to the enum.